lib/pbench/server/api/resources/users_api.py

```diff
@@ -284,24 +285,55 @@ def post(self):
                     }
         """
         auth_token = Auth.get_auth_token()
-        user = Auth.verify_auth(auth_token=auth_token)
+        state = Auth.verify_internal_token(auth_token)
 
-        # "None" user represents that either the token is not present in our database or it is an expired token.
-        # Expired token is already deleted by now if we reach here.
-        # In either case we will return suceess to a client, since the same token can not be used again.
+        # The only time a token won't be in the database is if the token is
+        # not a valid token.  Valid tokens that are not expired should be in
+        # the database, and expired tokens might or might not be in the
+        # database.  If the valid token is found in the database associated
+        # with the user, remove it.
```

Comment:
I think text needs some work, since the first sentence seems to be in contradiction with the second half of the second sentence.

I think what we want to say is something like
> If the token is invalid, it won't be in the database; valid but expired tokens may have been removed from the database by other requests; otherwise, the token should be in the database.  If the token is valid, found in the database, and associated with a user, then remove it from the database.

Reply:
Thanks, I'll reword.